File: reddio-csharp/reddio.api/V1/Rest/GetAssetIdMessage.cs
using Newtonsoft.Json;

namespace Reddio.Api.V1.Rest
{
    public class GetAssetIdMessage
    {
        [JsonProperty("contract_address")] public string ContractAddress { get; set; }
        [JsonProperty("type")] public string Type { get; set; }
        [JsonProperty("token_id")] public string TokenId { get; set; }
        [JsonProperty("quantum")] public long Quantum { get; set; }

        public GetAssetIdMessage()
        {
        }

        public GetAssetIdMessage(string contractAddress, string type, string tokenId)
        {
            ContractAddress = contractAddress;
            Type = type;
            TokenId = tokenId;
            Quantum = 1;
        }

        public GetAssetIdMessage(string contractAddress, string type, string tokenId, long quantum)
        {
            ContractAddress = contractAddress;
            Type = type;
            TokenId = tokenId;
            Quantum = quantum;
        }
    }
}

File: reddio-java/reddio-api/src/main/java/com/reddio/api/v1/rest/GetAssetIdMessage.java
package com.reddio.api.v1.rest;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor(staticName = "of")
public class GetAssetIdMessage {
    @JsonProperty("contract_address")
    private String contractAddress;
    @JsonProperty("type")
    private String type;
    @JsonProperty("token_id")
    private String tokenId;
    @JsonProperty("quantum")
    private Long quantum;
}
